webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/WebSitesRequestRouter.java

```diff
@@ -90,1056 +95,1093 @@ public WebSiteSessionController createComponentSessionController(
+    String tempPopup = request.getParameter(POPUP_PARAM);
+    String listeIcones = request.getParameter(ICON_LIST_PARAM);
+    String listeTopics = request.getParameter(TOPIC_LIST_PARAM);
+
+    // Retrieve positions
+    String positions = request.getParameter("Positions");
+    int popup = (tempPopup != null) && (tempPopup.length() > 0) ? 1 : 0;
+    List<String> listIcons =
+        listeIcones != null ? Arrays.asList(listeIcones.split(",")) : new ArrayList<>();
+    if (listIcons.get(0).isEmpty()) {
+      listIcons.remove(0);
+    }
+          /* recuperation de l'id */
+    String id = scc.getNextId();
 
-            // delete publication
-            String pubId = scc.getIdPublication(idToDelete);
-            scc.deletePublication(pubId);
+          /* Persist siteDetail inside database, type 1 = bookmark */
+    SiteDetail descriptionSite =
+        new SiteDetail(id, scc.getComponentId(), nomSite, description, nomPage, 1, null, null, 0,
+            popup);
+    descriptionSite.setPositions(positions);
 
-            begin = end + 1;
-            end = liste.indexOf(',', begin);
-          }
+    String pubId = scc.createWebSite(descriptionSite);
 
-          /* delete en BD */
-          scc.deleteWebSites(listToDelete);
-        } else if (action != null && "updateDescription".equals(action)) {
-
-          String id = request.getParameter("Id");
-          String nomSite = request.getParameter("nomSite");
-          String description = request.getParameter("description");
-          String nomPage = request.getParameter("nomPage");
-          String tempPopup = request.getParameter("popup");
-          String letat = request.getParameter("etat");
-          String listeIcones = request.getParameter("ListeIcones");
-          String listeTopics = request.getParameter("ListeTopics");
-
-          int popup = 0;
-          if ((tempPopup != null) && (tempPopup.length() > 0)) {
-            popup = 1;
-          }
+    if (listIcons.size() > 0) {
+      scc.associateIcons(id, listIcons);
+    }
 
-          int etat = -1;
-          if (StringUtil.isDefined(letat)) {
-            etat = Integer.parseInt(letat);
-          }
+    boolean publish = false;
+    for (String idTopic : listeTopics.split(",")) {
+      if (!idTopic.isEmpty()) {
+        scc.addPublicationToFolder(pubId, idTopic);
+        publish = true;
+      }
+    }
 
-          ArrayList<String> listIcons = new ArrayList<>();
-          int begin = 0;
-          int end = 0;
-          if (listeIcones != null) {
-            end = listeIcones.indexOf(',', begin);
-            while (end != -1) {
-              listIcons.add(listeIcones.substring(begin, end));
-              begin = end + 1;
-              end = listeIcones.indexOf(',', begin);
-            }
-          }
+    ArrayList<String> arrayToClassify = new ArrayList<>();
+    if (publish) {
+      arrayToClassify.add(id);
+      scc.publish(arrayToClassify); // set etat du site a 1
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3ATrailingCommentCheck)